src/lookup/derive.ts

```diff
@@ -60,56 +115,20 @@ export default async function getVerifiableCredentialAllFromShape(
   if (internalOptions.fetch === undefined) {
     internalOptions.fetch = fallbackFetch;
   }
-  // credentialClaims should contain all the claims, but not the context.
-  // const { "@context": claimsContext, ...credentialClaims } = vcShape;
-  // The following lines refactor the previous deconstruction in order to work
-  // around a misalignment between `rollup-plugin-typescript2` and NodeJS.
-  // Issue tracking: https://github.com/ezolenko/rollup-plugin-typescript2/issues/282
-  const credentialClaims = { ...vcShape };
-  delete credentialClaims["@context"];
-  const claimsContext = vcShape["@context"];
-  const credentialRequestBody: {
-    verifiableCredential: Partial<VerifiableCredential>;
-    options?: { include: typeof INCLUDE_EXPIRED_VC_OPTION };
-  } = {
-    // See https://w3c-ccg.github.io/vc-api/holder.html
-    verifiableCredential: {
-      "@context": concatenateContexts(defaultContext, claimsContext),
-      ...credentialClaims,
-    },
-  };
-  if (internalOptions.includeExpiredVc) {
-    credentialRequestBody.options = {
-      include: INCLUDE_EXPIRED_VC_OPTION,
-    };
-  }
-  const response = await internalOptions.fetch(holderEndpoint, {
-    headers: {
-      "Content-Type": "application/json",
-    },
-    method: "POST",
-    body: JSON.stringify(credentialRequestBody),
+  // The request payload depends on the target endpoint.
+  const vpRequest = holderEndpoint.endsWith("/query")
+    ? // The target endpoint is spec-compliant, and uses a standard VP request.
+      // This is based on an implementation-specific assumption about the endpoint
+      // being available under the /query path.
+      buildQueryByExample(vcShape)
+    : // The target endpoint is legacy, and uses a proprietary request format.
+      (buildLegacyQuery(
+        vcShape,
+        options?.includeExpiredVc ?? false
+        // The legacy proprietary format is casted as a VP request to be passed to the `query` function.
+      ) as unknown as VerifiablePresentationRequest);
```

Comment:
The reason this is done this way is because we know that the legacy request object is not of type `VerifiablePresentationRequest`. I moved all of the actual querying code to the `query` function, so this unsafe cast is a way to avoid code duplication because the alternative would be to issue the query directly from `getVerifiableCredentialAllFromShape`, instead of layering it on top of the lower-level `query` function.

I think having this cast is easier to maintain than duplicating the code, especially considering that we have unit tests to check for the well-formedness of the legacy request.